src/v/cluster/tx_gateway.cc

```diff
@@ -38,23 +38,31 @@ tx_gateway::init_tm_tx(init_tm_tx_request&&, rpc::streaming_context&) {
 }
 
 ss::future<begin_tx_reply>
-tx_gateway::begin_tx(begin_tx_request&&, rpc::streaming_context&) {
-    return ss::make_ready_future<begin_tx_reply>(begin_tx_reply());
+tx_gateway::begin_tx(begin_tx_request&& request, rpc::streaming_context&) {
+    return _rm_partition_frontend.local().do_begin_tx(request.ntp, request.pid);
```

Comment:
https://github.com/vectorizedio/redpanda/pull/1314#discussion_r624612252 also added a comment about it and an explicit leadership check (I didn't have it in the first place because it's anyway maybe become stale after the request execution started so we need to tolerate this)